### PR TITLE
add rw_shared switch to allow xclRegRead/xclRegWrite access shared CU

### PR DESCRIPTION
**Pull request by mamin506**
.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c [hunk 1]
```diff
@@ -4613,7 +4618,8 @@ int cu_map_addr(struct platform_device *pdev, u32 cu_idx, void *drm_filp,
 		mutex_unlock(&xdev->dev_lock);
 		return -EINVAL;
 	}
-	if (ip_excl_holder(exec, cu_idx) == 0) {
+	/* WORKAROUND: If rw_shared is true, allow map shared CU */
+	if (!exec->rw_shared && ip_excl_holder(exec, cu_idx) == 0) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c:4622]: do we want to check if user is root?

**Comment by stsoe** [comment]
Hi @mamin506 @maxzhen  It would be helpful with pull request comments that explain why this work-around is needed.  What's implemented here is easy to understand, but the why is missing.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c:4622]: We could easily add that once we have to.

**Comment by mamin506** [comment]
Hi @stsoe , this special requirement is from a specific customer. And it would be a non document option.

**Pull request merged**